modules/swagger-codegen/src/main/java/io/swagger/codegen/CliOption.java

```diff
@@ -20,4 +33,49 @@ public String getDescription() {
     public void setDescription(String description) {
         this.description = description;
     }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getDefault() {
+        return defaultValue;
+    }
+
+    public void setDefault(String defaultValue) {
+        this.defaultValue = defaultValue;
+    }
+
+    public CliOption addEnum(String value, String description) {
+        if (this.enumValues == null) {
+            this.enumValues = new LinkedHashMap<String, String>();
+        }
+        if (!enumValues.containsKey(value)) {
+            enumValues.put(value, description);
+        }
+        return this;
+    }
+
+    public Map<String, String> getEnum() {
+        return enumValues;
+    }
```

Comment:
In the java library, i see this value in the payload:

```json
  "library": {
    "opt": "library",
    "description": "library template (sub-template) to use:",
    "type": "enum",
    "enum": {
      "<default>": "HTTP client: Jersey client 1.18. JSON processing: Jackson 2.4.2",
      "jersey2": "HTTP client: Jersey client 2.6",
      "okhttp-gson": "HTTP client: OkHttp 2.4.0. JSON processing: Gson 2.3.1",
      "retrofit": "HTTP client: OkHttp 2.4.0. JSON processing: Gson 2.3.1 (Retrofit 1.9.0)"
    },
    "optionHelp": "library template (sub-template) to use:\n<default> - HTTP client: Jersey client 1.18. JSON processing: Jackson 2.4.2\njersey2 - HTTP client: Jersey client 2.6\nokhttp-gson - HTTP client: OkHttp 2.4.0. JSON processing: Gson 2.3.1\nretrofit - HTTP client: OkHttp 2.4.0. JSON processing: Gson 2.3.1 (Retrofit 1.9.0)"
  }
```

I believe we should see `string` for `type`, not `enum`.